### ISSUE TITLE
Unable to reset password through web

### ISSUE THREAD

**Issue by elightcap** [vector-im/element-web#23655]
### Steps to reproduce

1. go to https://app.element.io
2. click 'sign in'
3. 'click forgot password?'
4. enter email, new password, and confirm password
5. click 'send reset password link'
6. open link from email
7. click confirm 'confirm changing my password'
8. return to password reset screen and click 'I have verified my email address'

#### What did you expect?
Password reset to be completed
#### What happened instead?
Recieve error saying 'Failed to verify email address: make sure you clicked the link in the email.'

### Operating system

Debian Linux

### Browser information

firefox 104.0.1

### URL for webapp

https://app.element.io

### Application version

web app 

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by fmartingr** [comment]
Happening to me as well, endpoint response:

``` json
{
  "errcode": "M_UNKNOWN_TOKEN",
  "error": "Invalid access token passed.",
  "soft_logout": false
}
```

**Comment by Sp4rkR4t** [comment]
Exact same issue here, Firefox 106

**Comment by weeman1337** [comment]
Thanks for reporting the issue. I can easily reproduce it.

**Comment by t3chguy** [comment]
Regressed by https://github.com/matrix-org/matrix-js-sdk/pull/2719

**Comment by elightcap** [comment]
sorry, i saw this was closed with #2822 but im still running into the same issue. did i miss something?

**Comment by t3chguy** [comment]
The code hasn't been released yet.
